crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1969,6 +1977,7 @@ impl<'db> KnownInstanceType<'db> {
             return None;
         }
         match (module.name().as_str(), instance_name) {
+            ("typing" | "typing_extensions", "Any") => Some(Self::Any),
```

Comment:
While `Any` is available in `typing_extensions`, it is a simple unconditional re-export from `typing`; there is no Python version in which it is actually defined in `typing_extensions` (unlike the other symbols here.) (This is because it is the oldest of these special forms, and has always existed in the `typing` module.) So we don't ever need to recognize it from a _definition_ in `typing_extensions`; if someone imports it from `typing_extensions`, they're still getting the type from `typing` module.
```suggestion
            ("typing", "Any") => Some(Self::Any),
```